builder/vsphere/examples/supervisor/nginx-template.pkr.hcl

```diff
@@ -0,0 +1,33 @@
+# Note: this is an example template file to show you how to use the vsphere-supervisor builder
+# to deploy a VM with Nginx installed. You can use this file as a starting point for your own.
+
+source "vsphere-supervisor" "vm" {
+  image_name = "<Image name of the source VM, e.g. 'ubuntu-impish-21.10-cloudimg>'"
+  class_name = "<VM class that describes the virtual hardware settings, e.g. 'best-effort-large'>"
+  storage_class = "<Storage class that provides the backing storage for volume, e.g. 'wcplocal-storage-profile'>"
+  kubeconfig_path = "<Path to kubeconfig file of the Supervisor cluster, e.g. '/home/user/.kube/config'>"
+  supervisor_namespace = "<Namespace of the source VM in Supervisor cluster>"
+  source_name = "<Name of the source VM and its required objects to deploy>"
+  network_type = "<Network type of the source VM, e.g. 'nsx-t'>"
+  ssh_username = "<SSH username to set in the source VM>"
+  ssh_password = "<SSH password to set in the source VM>"
+  ssh_bastion_host = "<IP address of the bastion host for Packer to access the source VM>"
+  watch_source_timeout_sec = "<Timeout in seconds to wait for the source VM to be ready>"
+  keep_input_artifact = "<Whether to keep the created source VM and its other objects>"
+}
+
+build {
+  sources = ["source.vsphere-supervisor.example-vm"]
+  provisioner "shell" {
+    inline = [
+      "yum install -qy nginx",
```

Comment:
While this may be a little pedantic, the example is Ubuntu but the command here is the provisioner are for older versions of RHEL/derivative. Ubuntu would use apt and newer versions of RHEL/derivatives would use dnf.

Reply:
Makes sense. I have updated the commands here to match the example image.